cylc/flow/task_pool.py

```diff
@@ -1913,14 +1978,30 @@ def remove_tasks(self, items):
         """Remove tasks from the pool (forced by command)."""
         itasks, _, bad_items = self.filter_task_proxies(items)
         for itask in itasks:
+            # Spawn next occurrence of xtrigger sequential task.
+            if (
+                itask.is_xtrigger_sequential
+                and (
+                    itask.identity not in
+                    self.xtrigger_mgr.sequential_has_spawned_next
+                )
+            ):
+                self.xtrigger_mgr.sequential_has_spawned_next.add(
+                    itask.identity
+                )
+                self.spawn_to_rh_limit(
+                    itask.tdef,
+                    itask.tdef.next_point(itask.point),
+                    itask.flow_nums
+                )
             self.remove(itask, 'request')
         if self.compute_runahead():
             self.release_runahead_tasks()
         return len(bad_items)
 
     def _get_flow_nums(
             self,
-            flow: List[str],
+            flow: List[Union[str, int]],
```

Comment:
Why is this change needed?

Reply:
Not sure I quite remember, but it may well have been because how it's stored in the DB and how it is internally.. I'll reverse it and see what happens .. pretty sure something failed

Reply:
It'll need to be reversed in a bunch of places.

Reply:
Just wondering as I don't think you made any changes to how flow number are handled, on this branch?

Reply:
Have reverted everywhere I would have changed it (`scheduler.py`, `task_pool.py`) .. Let's see what happens, fast tests and xtirgger functional passed locally..

Reply:
All passed 🎉 - I guess we can go with the reversion and resolve this discussion?